packages/graph/hash_graph/lib/graph/src/store/postgres/knowledge/entity/mod.rs

```diff
@@ -46,31 +46,55 @@ impl<C: AsClient> PostgresStore<C> {
+        mut resolve_depths: GraphResolveDepths,
+        time_projection: TimeProjection,
     ) -> Pin<Box<dyn Future<Output = Result<(), QueryError>> + Send + 'a>> {
         async move {
-            let dependency_status = dependency_context
-                .knowledge_dependency_map
-                .insert(&entity_vertex_id, current_resolve_depth);
-
             let time_axis = subgraph.resolved_time_projection.image_time_axis();
 
-            let entity: &Entity = match dependency_status {
-                DependencyStatus::Unresolved => {
-                    match entity_vertex_id.subgraph_vertex_entry(subgraph) {
-                        RawEntryMut::Occupied(entry) => entry.into_mut(),
-                        RawEntryMut::Vacant(_) => {
-                            // Entities are always inserted into the subgraph before they are
-                            // resolved, so this should never happen. If it does, it is a bug.
-                            unreachable!("entity should already be in the subgraph")
-                        }
-                    }
+            let entity: &Entity = match entity_vertex_id.subgraph_vertex_entry(subgraph) {
+                RawEntryMut::Occupied(entry) => entry.into_mut(),
+                RawEntryMut::Vacant(_) => {
+                    // Entities are always inserted into the subgraph before they are resolved, so
+                    // this should never happen. If it does, it is a bug.
+                    unreachable!("entity should already be in the subgraph")
                 }
+            };
+
+            let version_interval = entity
+                .metadata()
+                .version()
+                .projected_time(time_axis)
+                .into_time_interval();
+
+            let Some(mut time_projection) = time_projection.intersect_image(version_interval) else {
+                // The entity is not in the time projection, so we don't need to resolve it.
+                // This should never happen as the time projection is based on this entity's
+                // version interval, but we emit a warning just in case.
+                tracing::warn!(
+                    entity_id = ?entity_vertex_id.base_id(),
+                    "entity is not in the time projection"
+                );
+                return Ok(());
+            };
 
+            let dependency_status = dependency_context.knowledge_dependency_map.update(
+                &entity_vertex_id,
+                resolve_depths,
+                time_projection.image(),
+            );
+
+            match dependency_status {
+                DependencyStatus::Unresolved(depths, interval) => {
+                    // The dependency may have to be resolved more than anticipated, so we update
+                    // the resolve depth and time projection.
+                    resolve_depths = depths;
+                    time_projection.set_image(interval);
```

Comment:
Generally, we should rely on the return value of `DependencyMap::update` to decide which parameters to use, but, yes, `depths` is always greater than or equal to `resolve_depths`. This may change with [this Asana task](https://app.asana.com/0/1201095311341924/1203763222255827/f) _(internal)_.